server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1306,6 +1306,34 @@ public <T extends RelationMetadata> List<T> relations(Class<T> clazz) {
         return relations;
     }
 
+    /**
+     * <p>
+     * Resolve the indices for a list of partitions and return their data either as
+     * {@link IndexMetadata} or derived from it using the {@code as} parameter.
+     * </p>
+     * <p>
+     * An empty list will resolve all indices in the cluster, open and closed
+     * </p>
+     * <p>
+     * {@code null} values returned from {@code as} are excluded from the result.
+     * This can be used to filter based on state or similar.
+     * </p>
+     * @param partitions A list of partitions to resolve
+     **/
+    public <T> List<T> getIndices(List<PartitionName> partitions, boolean strict, Function<IndexMetadata, T> as) {
+        if (partitions.isEmpty()) {
+            IndicesOptions indicesOptions = strict
+                ? IndicesOptions.STRICT_EXPAND_OPEN_CLOSED
+                : IndicesOptions.LENIENT_EXPAND_OPEN_CLOSED;
+            return mapIndices(IndexNameExpressionResolver.concreteIndices(this, indicesOptions), as);
```

Comment:
Given that `IndexNameExpressionResolver.concreteIndices` is deprecated, couldn't we here go through the `indices` keys/values directly?

Reply:
I was re-using the same logic as in `getIndices`, which also uses `IndexNameExpressionResolver.concreteIndices()`.  Let's replace both uses in a follow-up?

Reply:
Isn't the other one still required for BWC?